### PR TITLE
Interpreter: implement overflow flag support, clean up carry handling

### PR DESCRIPTION
**Pull request by FioraAeterna**
Loosely based on the approach used in SheepShaver's interpreter.

Probably not tested enough.

### CODE REVIEW

**Comment by skidau** [comment]
Have tested this out and it seems to work.  Would you please fix the android compile problem?

**Comment by JMC47** [comment]
This is required to make the Advance Game Port work on Interpreter.

File: Source/Core/Core/PowerPC/Interpreter/Interpreter.h [hunk 1]
```diff
@@ -297,7 +297,8 @@ class Interpreter : public CPUCoreBase
 	static void RunTable59(UGeckoInstruction _instCode);
 	static void RunTable63(UGeckoInstruction _instCode);
 
-	static u32 Helper_Carry(u32 _uValue1, u32 _uValue2);
+	static u32 Helper_Carry(u32 a, u32 b, u32 c);
```
Comment (degasus) [inline Source/Core/Core/PowerPC/Interpreter/Interpreter.h]: Maybe add a default value for c=0 for compatibility with the arm jit.

**Comment by lioncash** [comment]
I doubt it's a related problem, but with this, my instruction tests for ADDMEO (and some other 'o' instruction variants) will cause the interpreter to crash with this. You can find said tests [here](https://github.com/lioncash/DolphinPPCTests/blob/master/source/Integer.cpp#L184)

**Comment by kkkllleee** [comment]
If you can update this to work on android, the you are set.

**Comment by comex** [comment]
@dolphin-emu-bot rebuild

@FioraAeterna I really don't want to close this, since it seems easy to fix.

**Comment by shuffle2** [comment]
@lioncash are these tests in the dolphin repo now (yes i'm lazy)? if not, why not?

**Comment by lioncash** [comment]
The tests are here: https://github.com/lioncash/DolphinPPCTests/blob/master/source/Integer.cpp

While testing this branch, my tests will never actually finish. It enters an infinite loop (or what seems to be one). However, without this merged, the tests do finish and just throw up PanicAlert dialogs whenever the overflow variants are hit; this leads me to suspect there's a bug somewhere in here

**Comment by shuffle2** [comment]
I see, so the tests are not merged because of the PanicAlerts (causing the test infrastructure to fail)?

**Comment by lioncash** [comment]
No, they aren't merged because these aren't using the test infrastructure, since the way it works is it's run on the console, generating a large output file of correct results. Then, it's run in the emulator and outputs another list; the two lists are compared (via diffing) and any differences on Dolphin's side shows inaccuracies. Dolphin's testing framework isn't capable of doing that.

**Comment by delroth** [comment]
No activity on the PR and @FioraAeterna is not actively contributing anymore. Closing.

**Pull request closed without merging**